File: src/mvc/cpp/emoji_picker/main_window.cpp
#include "main_window.h"
#include "./ui_main_window.h"
#include <QDir>
#include <QFileInfo>
#include <QModelIndex>

MainWindow::MainWindow(QWidget *parent)
    : QMainWindow(parent), ui(new Ui::MainWindow) {
  ui->setupUi(this);

  auto model = new EmojiPickerModel(this);

  QStringList iconPaths;

  // find all paths in images folder
  QDir dir("images");
  QFileInfoList fileInfoList = dir.entryInfoList(QDir::Files);
  for (auto fileInfo : fileInfoList) {
    iconPaths.append(fileInfo.absoluteFilePath());
  }
  QList<EmojiData> emojis;

  for (const auto &iconPath : iconPaths) {
    emojis.append(EmojiData(iconPath, iconPath, QList<QString>()));
  }

  EmojiCategory category = EmojiCategory("Category A", emojis);
  model->setColumnCount(7);
  model->appendRow(category);
  category.name = "Category B";
  model->appendRow(category);

  ui->treeView->setModel(model);
}

MainWindow::~MainWindow() { delete ui; }

File: src/mvc/python/emoji_picker/main_window.py
from emoji_picker_data import EmojiCategory, EmojiData
from emoji_picker_tree import EmojiPickerModel
from PyQt6 import QtWidgets, uic
from PyQt6.QtCore import QDir


class MainWindow(QtWidgets.QMainWindow):
    def __init__(self):
        super().__init__()
        uic.loadUi("main_window.ui", self)
        model = EmojiPickerModel(self)

        iconPaths = []

        # find all paths in images folder
        dir = QDir("images")
        fileInfoList = dir.entryInfoList(QDir.Filter.Files)
        for fileInfo in fileInfoList:
            iconPaths.append(fileInfo.absoluteFilePath())

        emojis = []

        for iconPath in iconPaths:
            emojis.append(EmojiData(iconPath, iconPath, list()))

        category = EmojiCategory("Category A", emojis)
        model.setColumnCount(7)
        model.appendRow(category)
        category.name = "Category B"
        model.appendRow(category)

        self.treeView.setModel(model)

        self.show()
